document.go

```diff
@@ -22,10 +37,14 @@ type Document struct {
 	} `json:"jsonapi"`
 	// Status is an HTTP Status Code
 	Status int `json:"-"`
+	// DataMode to enforce for the document
+	Mode DocumentMode `json:"-"`
```

Comment:
I am making this public mainly to allow for 3rd party applications (JSC as an example) to set these manually. The most probable use case is setting ErrorMode when receiving a response to a sent request.